test/src/sdk-harness/test_projects/storage_vec/testgen.rs

```diff
@@ -320,6 +376,121 @@ macro_rules! testgen {
 
                     assert!(is_empty(&instance).await);
                 }
+
+                #[tokio::test]
+                async fn can_swap() {
+                    let instance = get_contract_instance().await;
+
+                    push(&instance, $arg0).await;
+                    push(&instance, $arg1).await;
+                    push(&instance, $arg2).await;
+                    push(&instance, $arg3).await;
+
+                    swap(&instance, 0, 3).await;
+                    swap(&instance, 1, 2).await;
+
+                    assert_eq!(get(&instance, 0).await, $arg3);
+                    assert_eq!(get(&instance, 1).await, $arg2);
+                    assert_eq!(get(&instance, 2).await, $arg1);
+                    assert_eq!(get(&instance, 3).await, $arg0);
+                }
+
+                #[tokio::test]
+                async fn can_get_first() {
+                    let instance = get_contract_instance().await;
+
+                    push(&instance, $arg0).await;
+                    push(&instance, $arg1).await;
+
+                    assert_eq!(first(&instance).await, $arg0);
+                }
+
+                #[tokio::test]
+                async fn can_get_last() {
+                    let instance = get_contract_instance().await;
+
+                    push(&instance, $arg0).await;
+                    push(&instance, $arg1).await;
+
+                    assert_eq!(last(&instance).await, $arg1);
+                }
+
+                #[tokio::test]
+                async fn can_reverse() {
```

Comment:
Can we add a test here for when the length of the vector is odd?

Reply:
done